docs/ldml/tr35.md

```diff
@@ -2904,17 +2920,67 @@ The syntax characters are listed in the table below:
 |  }   | U+007D | RIGHT CURLY BRACKET  | Strings in set; Perl property syntax       |
 |      | U+0020 U+0009..U+000D U+0085<br/>U+200E U+200F<br/>U+2028 U+2029 | ASCII whitespace,<br/>LRM, RLM,<br/>LINE/PARAGRAPH SEPARATOR | Ignored except when escaped |
 
+Note that some syntax characters only have a special meaning in a certain context. In particular:
+* Out of all above syntax characters, only \\, \}, and whitespace have a special meaning inside strings (**\[\{\[a-z\]\}\]** is the set of the string '\[a-z\]', **\[\{\$blah\}\]** is the set of the string '\$blah').
+* \$ is equivalent to \uFFFF when appearing at the very end of a set with or without trailing whitespace (**[a-z\$]**, **[a-z\$ ]**), and used as starting indicator for a variable reference elsewhere, in which case the variable name will be the longest match on the `var` nonterminal (**[\$my_set]**).
+* \- is equivalent to the literal character \\- when occuring at the very beginning of a set, after a \^ at the beginning of a set, or at the very end of a set, in all cases with or without whitespace (**[-abc]**, **[ ^ -abc]**, **[abc-]**), and used as the set difference or range operator elsewhere (**[[abc]-[bc]]**, **[a-z]**)
+* \: initiates a POSIX property set when directly after a \[ without whitespace inbetween (**[:L:]**), ends a POSIX property set when directly before a \] without whitespace inbetween (**[:L:]**), and is equivalent to the literal character \\\: in any other place (**[ \:]**, **[L\:]**)
+* \} ends a string when occurring inside a string (**[{hello}]**), and is equivalent to the literal character \\\} in any other place (**[}a]**)
+
+###### Syntax Special Case Examples
+In the following, a table of examples including common sources of confusion concerning the UnicodeSet syntax:
```

Comment:
Had an idea for reorging this for clarity, see https://docs.google.com/spreadsheets/d/1xbLH2pTFHR9AG0H3Ybl3aNsoxivzOr2Y_OcVUlYn_8M/edit

Reply:
Thanks, implemented the changes!